File: NetFramework/C1.Win.Ribbon/VB/WordPad/BackstageTabs/OpenTab.vb
Public Class OpenTab

    Public Sub New()
        Initilize()
    End Sub

    Protected Overrides Sub OnMenuItemClick(ByVal mi As ListItem)
        Owner.OpenDocument(System.IO.Path.Combine(mi.Item.SubText, mi.Item.Text))
        mi.Item.LDate = DateTime.Now
    End Sub

    Protected Overrides Sub OnMenuButtonClick()
        Owner.OpenDocument()
    End Sub

    Private Sub OpenTab_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Button.ItemImage = My.Resources.Open
        Button.ItemText = My.Resources.Browse_txt
        Caption = GetElementText(My.Resources.OpenTab_txt)
    End Sub
End Class

File: NetFramework/Ribbon/CS/WordPad/BackstageTabs/OpenTab.cs
using System;
using WordPad.BackstageTabs.Items;

namespace WordPad.BackstageTabs
{
    public partial class OpenTab : BackstageTab
    {
        public OpenTab()
        {
            Button.ItemImage = Properties.Resources.Open;
            Button.ItemText = Properties.Resources.Browse_txt;
            Caption = GetElementText(Properties.Resources.OpenTab_txt);
        }

        #region override

        protected override void OnMenuItemClick(ListItem mi)
        {
            Owner.OpenDocument(System.IO.Path.Combine(mi.Item.SubText, mi.Item.Text));
            mi.Item.Date = DateTime.Now;
        }

        protected override void OnMenuButtonClick()
        {
            Owner.OpenDocument();
        }

        #endregion        
    }
}
